libparsec/crates/serialization_format/src/protocol.rs

```diff
@@ -171,7 +174,7 @@ struct GenCmdsFamily {
 //
 
 impl GenCmdsFamily {
-    fn new(cmds: Vec<JsonCmd>, family_name: &str) -> Self {
+    fn new(cmds: Vec<JsonCmd>, family_name: &str, reuse_schemas: bool) -> Self {
```

Comment:
```suggestion
    fn new(cmds: Vec<JsonCmd>, family_name: &str, reuse_schemas: ReuseSchemaStrategy) -> Self {
```

with:
```rust
enum ReuseSchemaStrategy {
  Default,
  Never,
}
```

I think it's more readable because we cannot "force always reuse schemas", we can only "force never reuse schema".
Hence `reuse_schemas=true` is the default behavior, and `reuse_schemas=false` the one setting a special behavior. So we should have instead something like `force_disable_reuse_schemas=true` or `never_reuse_schemas=true`.

But I find it a bit odd to have a boolean set to true to disable something, it's just more readable to have a specific enum to precisely explain what we are going to do

Reply:
It is indeed better. I also added some docstrings.